### PR TITLE
Allows voice changers to properly mimick robotic voices

### PR DESCRIPTION
**Pull request by ShiftyRail**
Say code is a mess. Half of the procs in its `readme` aren't even used anymore except in the very edge case the mob doesn't speak a language.

This PR fixes that by adding helpful documentation so that future coders (if they even exist) won't be as confused as I was.

It also allows people wearing a voice changer to properly mimic silicons.

[bugfix]
:cl:
- rscadd: The syndicate voice changer can now properly mimic silicons.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
>choice changers

File: code/modules/mob/living/carbon/human/say.dm [hunk 1]
```diff
@@ -21,6 +24,20 @@
 
 	return "says, [text]";
 
+// Use this for an override of the spoken verb.
+/mob/living/carbon/human/get_spoken_verb(var/msg)
+	if(istype(wear_mask, /obj/item/clothing/mask/gas/voice))
```
Comment (jknpj) [inline code/modules/mob/living/carbon/human/say.dm]: and the cardborg helmet?

File: code/modules/clothing/masks/gasmask.dm [hunk 2]
```diff
@@ -122,6 +123,21 @@
...
+	var/obj/item/clothing/mask/gas/voice/T = target
+	if(!istype(T))
+		return
+	switch (T.speech_mode)
+		if (VOICE_CHANGER_SAYS)
+			T.speech_mode = VOICE_CHANGER_STATES
+			to_chat(owner, "<span class='notice'>You will now <i>state</i> things like a machine would.</span>")
+		if (VOICE_CHANGER_STATES)
+			T.speech_mode = VOICE_CHANGER_SAYS
+			to_chat(owner, "<span class='notice'>You will now <i>say</i> things like a machine would.</span>")
```
Comment (boy2mantwicethefam) [inline code/modules/clothing/masks/gasmask.dm]: "You will now say things like a machine would"

**Comment by Kurfursten** [comment]
how are you going to imitate a borg name in the first place? no ID

**Comment by ShiftyRail** [comment]
@kurf syndicate agent ID with a custom name.

File: code/modules/mob/living/carbon/human/say.dm [hunk 3]
```diff
@@ -21,6 +24,22 @@
 
 	return "says, [text]";
 
+// Use this for an override of the spoken verb.
+/mob/living/carbon/human/get_spoken_verb(var/msg)
+	if(istype(wear_mask, /obj/item/clothing/mask/gas/voice) || istype(wear_mask, /obj/item/clothing/head/cardborg/))
```
Comment (jknpj) [inline code/modules/mob/living/carbon/human/say.dm]: stray /

**Review by SonixApache — COMMENTED** [review]
delete this commit title

**Pull request merged**